### PR TITLE
Use human readable string instead of the integer when printing flush/compaction reason

### PR DESCRIPTION
**Pull request by git-hulk**
Before applying this patch:

```
column family: metadata, job_id: 59, compaction reason: 1, output compression type: no
```

And after this patch:

```
column family: metadata, job_id: 59, compaction reason: LevelMaxLevelSize, output compression type: no
```

### CODE REVIEW

**Comment by git-hulk** [comment]
Need to hold this PR till the speedb supports those functions.

File: src/storage/event_listener.cc [hunk 1]
```diff
@@ -88,9 +99,25 @@ void EventListener::OnCompactionCompleted(rocksdb::DB *db, const rocksdb::Compac
...
 }
 
+void EventListener::OnSubcompactionBegin(const rocksdb::SubcompactionJobInfo &si) {
+  LOG(INFO) << "[event_listener/subcompaction_begin] column family: " << si.cf_name << ", job_id: " << si.job_id
+            << ", compaction reason: " << rocksdb::GetCompactionReasonString(si.compaction_reason)
+            << ", output compression type: " << CompressType2String(si.compression);
+}
+
+void EventListener::OnSubcompactionCompleted(const rocksdb::SubcompactionJobInfo &si) {
+  LOG(INFO) << "[event_listener/subcompaction_begin] column family: " << si.cf_name << ", job_id: " << si.job_id
```
Comment (PragmaTwice) [inline src/storage/event_listener.cc]:
```suggestion
  LOG(INFO) << "[event_listener/subcompaction_completed] column family: " << si.cf_name << ", job_id: " << si.job_id
```

File: src/storage/event_listener.cc [hunk 2]
```diff
@@ -75,9 +75,20 @@ bool IsDiskQuotaExceeded(const rocksdb::Status &bg_error) {
...
 
+void EventListener::OnCompactionBegin(rocksdb::DB *db, const rocksdb::CompactionJobInfo &ci) {
+  LOG(INFO) << "[event_listener/compaction_begin] column family: " << ci.cf_name << ", job_id: " << ci.job_id
+            << ", compaction reason: " << rocksdb::GetCompactionReasonString(ci.compaction_reason)
+            << ", output compression type: " << CompressType2String(ci.compression)
+            << ", base input level(files): " << ci.base_input_level << "(" << ci.input_files.size() << ")"
+            << ", output level(files): " << ci.output_level << "(" << ci.output_files.size() << ")"
+            << ", input bytes: " << ci.stats.total_input_bytes << ", output bytes:" << ci.stats.total_output_bytes
+            << ", is_manual_compaction:" << (ci.stats.is_manual_compaction ? "yes" : "no")
+            << ", elapsed(micro): " << ci.stats.elapsed_micros;
```
Comment (mapleFU) [inline src/storage/event_listener.cc]: what would `ci.stats.elapsed_micros` in begin means?

Comment (git-hulk) [reply src/storage/event_listener.cc]: That's meaningless for the begin function, so let's remove it.

Comment (git-hulk) [inline src/storage/event_listener.cc]:
```suggestion
            << ", is_manual_compaction:" << (ci.stats.is_manual_compaction ? "yes" : "no");
```

**Review by mapleFU — APPROVED** [review]
+1

**Review by PragmaTwice — APPROVED** [review]

**Pull request merged**